lib/vtls/gskit.c

```diff
@@ -978,9 +979,8 @@ static CURLcode gskit_connect_step2(struct Curl_easy *data,
     timediff_t timeout_ms = nonblocking? 0: Curl_timeleft(data, NULL, TRUE);
     if(timeout_ms < 0)
       timeout_ms = 0;
-    stmv.tv_sec = timeout_ms / 1000;
-    stmv.tv_usec = (timeout_ms - stmv.tv_sec * 1000) * 1000;
-    switch(QsoWaitForIOCompletion(BACKEND->iocport, &cstat, &stmv)) {
+    switch(QsoWaitForIOCompletion(BACKEND->iocport, &cstat,
+                                  curlx_mstotv(timeout_ms))) {
```

Comment:
Yes. IMO it's clearer to init smtv to 0 , call curlx_mstotv and pass &smtv as the param because that is always !=NULL but this would work as well:
```suggestion
                                  curlx_mstotv(&stmv, timeout_ms))) {
```
this one though there's an assumption that curlx_mstotv will always return != NULL. though that may be the case now code could change.

Reply:
yes, but with your first suggestion you assume that curlx_mstotv always works on the timeval struct provided. Right now the idea is to use the return value which at the time of this writing is based on the first argument, but this could also change at some point.